spec/pager_spec.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+require 'array_database'
+require 'pager'
+require 'user_interface'
+require 'validator'
+
+RSpec.describe Pager do
+  let(:database) { ArrayDatabase.new }
+  let(:input) { StringIO.new }
+  let(:output) { StringIO.new }
+  let(:pager) { Pager.new(user_interface, database) }
+  let(:user_interface) { UserInterface.new(input, output, validator) }
+  let(:validator) { Validator.new }
+
+  describe '#run' do
+    it 'prints a message to the user if the database is empty' do
+      pager.run
+
+      expect(output.string).to eq(UserInterface::NO_CONTACTS_MESSAGE)
+    end
+
+    it 'does not prints a message if the database has any contacts' do
+      database.create({ name: 'Matt Damon' })
+
+      pager.run
+
+      expect(output.string).to_not eq(UserInterface::NO_CONTACTS_MESSAGE)
+    end
+
+    it 'sorts contacts into aphabetical order according to name' do
+      database.create({ name: 'Warran Smith' })
+      database.create({ name: 'Matt Damon' })
+
+      pager.run
+
+      expect(output.string).to eq(letter_header('M') +
```

Comment:
If possible I would use `match` and a regex here, because otherwise you have to repeat the same code you have in the user interface.

Also in these tests that test the order, I would check the initial, as it is not relevant for the test, and you have dedicated tests for the initial.